tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/SwaggerAPIViewGenerator.cs

```diff
@@ -124,7 +124,16 @@ public static SwaggerApiViewSpec GenerateSwaggerApiView(SwaggerSpec swaggerSpec,
                         
                         if (response.schema != null)
                         {
-                            schema = schemaCache.GetResolvedSchema(schema, currentSwaggerFilePath);
+                            // The initial refChain is the root level schema.
+                            // There are some scenarios that the property of the root level schema is a ref to the root level itself (circular reference).
+                            // Like "errorDetail" schema in common types.
+                            LinkedList<string> refChain = new LinkedList<string>();
+                            if (schema.Ref != null)
+                            {
+                                refChain.AddFirst(schema.Ref);
+                            }
+                            
+                            schema = schemaCache.GetResolvedSchema(schema, currentSwaggerFilePath, refChain);
```

Comment:
Do we need this change here as well? https://github.com/Azure/azure-sdk-tools/blob/ffb60fcabb8ea12b22c8940d3273b87d45b29a59/tools/apiview/parsers/swagger-api-parser/SwaggerApiParser/SwaggerAPIViewGenerator.cs#L94

Reply:
I think maybe for now, we don't need added here. Since I don't see any parameters has circular reference in spec repo.